runtime/vm/createramclass.cpp

```diff
@@ -2872,16 +2921,20 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 
 	if (!loadSuperClassAndInterfaces(vmThread, hostClassLoader, romClass, elementClass, packageID, hotswapping, classPreloadFlags, &superclass, module)
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-		|| !loadFlattenableFieldValueClasses(vmThread, hostClassLoader, romClass, classPreloadFlags, packageID, module)
+		|| !loadFlattenableFieldValueClasses(vmThread, hostClassLoader, romClass, classPreloadFlags, packageID, module, &largestFieldType, &flattenableInstanceFieldCount)
```

Comment:
Haven't all the classes been loaded already by the field offset walk? I'd rather see the VT classes loaded ASAP and then not pass the loader to the field offset walk (so that it cannot fail).